src/Symfony/Component/Cache/Simple/AbstractCache.php

```diff
@@ -182,7 +182,7 @@ private function generateValues(iterable $values, array &$keys, $default): itera
         try {
             foreach ($values as $id => $value) {
                 if (!isset($keys[$id])) {
-                    $id = key($keys);
+                    throw new InvalidArgumentException(sprintf('Could not match value id "%s" to keys "%s".', $id, implode('", "', $keys)));
```

Comment:
This is what lead to the mix up of values. The assumption here is: if the id of the value cannot be found in the original keys, let’s just assume it to be the first one. Instead of returning a wrong value, we should create a miss here.

Reply:
:+1:

Reply:
I realize I wasn’t precise in my language: this is not the root cause of the issue but the subsequent error that lead to values being mixed up. The root cause was that they version ID contained a slash, that slash was replaced with a null byte in `decodeKey` and therefore the keys no longer matched and the piece of code above was triggered.

Reply:
FYI, we missed patching this other file, now fixed:
https://github.com/symfony/symfony/commit/25941ffe732461fd4ff0f567153ee6645c8df7eb